From 7384f79f69fdb7b691cc5b0c28c301b3fe8b633e Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Thu, 26 May 2016 10:46:57 +0200
Subject: [PATCH] xf86atomic: require CAS support in libatomic_ops

Since AO_compare_and_swap_full() is used by libdrm, AO_REQUIRE_CAS
must be defined before including <atomic_ops.h> so that we are sure
that CAS support will be provided. This is necessary to make sure that
the AO_compare_and_swap_full() function will be provided on all
architectures, including the ones that don't have built-in CAS support
such as SPARCv8.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 xf86atomic.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xf86atomic.h b/xf86atomic.h
index 922b37d..d7017a5 100644
--- a/xf86atomic.h
+++ b/xf86atomic.h
@@ -58,6 +58,7 @@ typedef struct {
 #endif
 
 #if HAVE_LIB_ATOMIC_OPS
+#define AO_REQUIRE_CAS
 #include <atomic_ops.h>
 
 #define HAS_ATOMIC_OPS 1
-- 
2.7.4

